### ISSUE TITLE
{{else if foo}} is not handled correctly

### ISSUE THREAD

**Issue by martndemus** [rwjblue/ember-cli-template-lint#36]
```handlebars
{{#if isMorning}}
  Good morning
{{else if isAfternoon}}
  Good afternoon
{{else}}
  Good night
{{/if}}
```
Will report:

```
Incorrect indentation for `if` beginning at ('rwe/components/criteria-operator-select/template.hbs'@ L3:C0).
Expected `{{/if}}` ending at ('rwe/components/criteria-operator-select/template.hbs'@ L7:C0)to be at an indentation of 0 but was found at -7.
```

**Comment by rwjblue** [comment]
I need to dig in a bit more on how the AST looks for the inverse blocks in general, I'm guessing that it thinks `-7` because `else if` is 7 chars but that is just a WAG...

**Comment by urbany** [comment]
Also, don't forget ``{{else unless x}} ``
(I think that is valid right?)

**Comment by urbany** [comment]
Hi @rwjblue I'm not sure if this was correctly fixed, I'm, getting an error here:
```handlebars
<div>
  {{#if isSingleError}}
    {{error}}
  {{else if errors}}
    {{errors}}
  {{else}}
    {{t 'no errors found'}}
  {{/if}}
</div>
```

```
Incorrect indentation for `div` beginning at ('project/templates/components/show-errors.hbs'@ L2:C0). Expected `</div>` ending at ('project/templates/components/show-errors.hbs'@ L10:C8)to be at an indentation of 0 but was found at 2.
```

If I remove the ``else if`` from the template I get no errors so I guess it must be connected to this issue:
```handlebars
<div>
  {{#if isSingleError}}
    {{error}}
  {{else}}
    {{t 'no errors found'}}
  {{/if}}
</div>
```

````
No errors
```

**Comment by rwjblue** [comment]
OK, thanks for the detailed examples, that really helped me track things down.

The issue with:

```hbs
    <div>
      {{#if isSingleError}}
        Single error here!
      {{else if errors}}
        Multiple errors here!
      {{else}}
        No errors found!
      {{/if}}
    </div>
```

Is due to a bug in HTMLBars parsing, which is fixed by https://github.com/tildeio/htmlbars/pull/450.

The other issue is just a silly hard coding of the number of `_startOffset` nodes (I hard coded to `7` but should have used `5 + inverseBlock.path.original.length` instead).  This can be fixed shortly...

**Comment by rwjblue** [comment]
**Bug 2** from https://github.com/rwjblue/ember-cli-template-lint/issues/36#issuecomment-193695086 was fixed by https://github.com/rwjblue/ember-cli-template-lint/pull/43 and released as v0.2.3.
